libraries/lib-music-information-retrieval/MirDsp.cpp

```diff
@@ -25,6 +25,21 @@ namespace MIR
 {
 namespace
 {
+constexpr float GetLog2(float x)
+{
+   // https://stackoverflow.com/a/28730362
+   union
+   {
+      float val;
+      int32_t x;
+   } u = { x };
+   auto log_2 = (float)(((u.x >> 23) & 255) - 128);
+   u.x &= ~(255 << 23);
+   u.x += 127 << 23;
+   log_2 += ((-0.3358287811f) * u.val + 2.0f) * u.val - 0.65871759316667f;
+   return log_2;
```

Comment:
I wrote elsewhere how I almost understand where the magic numbers come from, and how the method might be generalized to higher or lower degree polynomial approximations.  We might experiment with the tradeoff of computation speed and accuracy of the log2 approximation.  But then the greatest gain is likely just from avoiding the transcendental function.

Reply:
I think so too. I must admit I have understood little of your demonstration, especially the bit-shift part, maybe you can explain that to me more at length in an upcoming conversation. Besides me not understanding it looked correct, judging by your results and also some I ran on my side.